poetry/console/commands/add.py

```diff
@@ -152,29 +152,28 @@ def handle(self):
 
             poetry_content[section][_constraint["name"]] = constraint
 
-        # Write new content
-        self.poetry.file.write(content)
+        try:
+            # Write new content
+            self.poetry.file.write(content)
 
-        # Cosmetic new line
-        self.line("")
+            # Cosmetic new line
+            self.line("")
 
-        # Update packages
-        self.reset_poetry()
+            # Update packages
+            self.reset_poetry()
 
-        self._installer.set_package(self.poetry.package)
-        self._installer.dry_run(self.option("dry-run"))
-        self._installer.verbose(self._io.is_verbose())
-        self._installer.update(True)
-        if self.option("lock"):
-            self._installer.lock()
+            self._installer.set_package(self.poetry.package)
+            self._installer.dry_run(self.option("dry-run"))
+            self._installer.verbose(self._io.is_verbose())
+            self._installer.update(True)
+            if self.option("lock"):
+                self._installer.lock()
 
-        self._installer.whitelist([r["name"] for r in requirements])
+            self._installer.whitelist([r["name"] for r in requirements])
 
-        try:
             status = self._installer.run()
-        except Exception:
+        except BaseException:
```

Comment:
Maybe add a comment here so we retain why we made this change.

Reply:
added!